authz/rbac_translator_test.go

```diff
@@ -0,0 +1,229 @@
+										{Identifier: &v3rbacpb.Principal_Authenticated_{
+											Authenticated: &v3rbacpb.Principal_Authenticated{PrincipalName: &v3matcherpb.StringMatcher{
+												MatchPattern: &v3matcherpb.StringMatcher_Prefix{Prefix: ""}}}}},
+									}}}}},
+						}}}},
+					Permissions: []*v3rbacpb.Permission{
+						{Rule: &v3rbacpb.Permission_AndRules{AndRules: &v3rbacpb.Permission_Set{
+							Rules: []*v3rbacpb.Permission{
+								{Rule: &v3rbacpb.Permission_OrRules{OrRules: &v3rbacpb.Permission_Set{
+									Rules: []*v3rbacpb.Permission{
+										{Rule: &v3rbacpb.Permission_UrlPath{
+											UrlPath: &v3matcherpb.PathMatcher{Rule: &v3matcherpb.PathMatcher_Path{Path: &v3matcherpb.StringMatcher{
+												MatchPattern: &v3matcherpb.StringMatcher_Prefix{Prefix: "path-foo"}}}}}},
+									}}}}}}}}},
+				},
+				"authz_allow_policy_2": {
+					Principals: []*v3rbacpb.Principal{
+						{Identifier: &v3rbacpb.Principal_Any{Any: true}}},
+					Permissions: []*v3rbacpb.Permission{
+						{Rule: &v3rbacpb.Permission_AndRules{AndRules: &v3rbacpb.Permission_Set{
+							Rules: []*v3rbacpb.Permission{
+								{Rule: &v3rbacpb.Permission_OrRules{OrRules: &v3rbacpb.Permission_Set{
+									Rules: []*v3rbacpb.Permission{
+										{Rule: &v3rbacpb.Permission_UrlPath{
+											UrlPath: &v3matcherpb.PathMatcher{Rule: &v3matcherpb.PathMatcher_Path{Path: &v3matcherpb.StringMatcher{
+												MatchPattern: &v3matcherpb.StringMatcher_Exact{Exact: "path-bar"}}}}}},
+										{Rule: &v3rbacpb.Permission_UrlPath{
+											UrlPath: &v3matcherpb.PathMatcher{Rule: &v3matcherpb.PathMatcher_Path{Path: &v3matcherpb.StringMatcher{
+												MatchPattern: &v3matcherpb.StringMatcher_Suffix{Suffix: "baz"}}}}}},
+									}}}},
+								{Rule: &v3rbacpb.Permission_AndRules{AndRules: &v3rbacpb.Permission_Set{
+									Rules: []*v3rbacpb.Permission{
+										{Rule: &v3rbacpb.Permission_OrRules{OrRules: &v3rbacpb.Permission_Set{
+											Rules: []*v3rbacpb.Permission{
+												{Rule: &v3rbacpb.Permission_Header{
+													Header: &v3routepb.HeaderMatcher{
+														Name: "key-1", HeaderMatchSpecifier: &v3routepb.HeaderMatcher_ExactMatch{ExactMatch: "foo"}}}},
+												{Rule: &v3rbacpb.Permission_Header{
+													Header: &v3routepb.HeaderMatcher{
+														Name: "key-1", HeaderMatchSpecifier: &v3routepb.HeaderMatcher_SuffixMatch{SuffixMatch: "bar"}}}},
+											}}}},
+										{Rule: &v3rbacpb.Permission_OrRules{OrRules: &v3rbacpb.Permission_Set{
+											Rules: []*v3rbacpb.Permission{
+												{Rule: &v3rbacpb.Permission_Header{
+													Header: &v3routepb.HeaderMatcher{
+														Name: "key-2", HeaderMatchSpecifier: &v3routepb.HeaderMatcher_PrefixMatch{PrefixMatch: "baz"}}}},
+											}}}}}}}}}}}}},
+				},
+			}},
+		},
+		"missing name field": {
+			authzPolicy:     `{}`,
+			wantErr:         "\"name\" is not present",
+			wantDenyPolicy:  nil,
+			wantAllowPolicy: nil,
```

Comment:
No need to initialize fields to their zero value.  (And if `wantErr` is set, it should be obvious that we don't want any policies.)